### PR TITLE
Add UDP module `broadcast`, `multicast` and `ttl`.

### PR DESCRIPTION
**Pull request by chokobole**
libtuv-DCO-1.0-Signed-off-by: wonyong.kim wonyong.kim@samsung.com

### CODE REVIEW

File: source/unix/uv_unix.c [hunk 1]
```diff
@@ -188,3 +188,45 @@ int uv_ip4_addr(const char* ip, int port, struct sockaddr_in* addr) {
...
+
+  memset(addr, 0, sizeof(*addr));
+  addr->sin6_family = AF_INET6;
+  addr->sin6_port = htons(port);
+
+  zone_index = strchr(ip, '%');
+  if (zone_index != NULL) {
+    address_part_size = zone_index - ip;
+    if (address_part_size >= sizeof(address_part))
+      address_part_size = sizeof(address_part) - 1;
```
Comment (zherczeg) [inline source/unix/uv_unix.c]: This is a silent limit (without configuration). Is this intended?

File: source/unix/uv_unix_udp.c [hunk 2]
```diff
@@ -531,25 +619,38 @@ int uv_udp_open(uv_udp_t* handle, uv_os_sock_t sock) {
...
+    err = uv__udp_maybe_deferred_bind(handle, AF_INET, UV_UDP_REUSEADDR);
+    if (err)
+      return err;
+    return uv__udp_set_membership4(handle, &addr4, interface_addr, membership);
+  } else if (uv_ip6_addr(multicast_addr, 0, &addr6) == 0) {
+    err = uv__udp_maybe_deferred_bind(handle, AF_INET6, UV_UDP_REUSEADDR);
+    if (err)
+      return err;
+    return uv__udp_set_membership6(handle, &addr6, interface_addr, membership);
+  } else {
```
Comment (zherczeg) [inline source/unix/uv_unix_udp.c]: else is no need here

Comment (chokobole) [inline source/unix/uv_unix_udp.c]: @zherczeg  I removed redundant else clause!

**Comment by zherczeg** [comment]
LGTM

**Comment by yichoi** [comment]
LGTM

**Pull request merged**